### PR TITLE
looks like rpc spec returns {badrpc, Reason}. 

### PR DESCRIPTION
**Pull request by linearregression**
* remove uneeded spec type since support only tcp now.
* type spec for F in MFA. For anonymous function the type is function. but for exported funciton it is atom()
* rpc returns Result | {badrpc, Reason}. 
  On exception, they are not always returned as {badrpc, Reason}.
  Want to bypass gen_rpc call if targetNode is itself?

### CODE REVIEW

File: include/app.hrl [hunk 1]
```diff
@@ -16,3 +16,5 @@
         {keepalive,true}, % Keep our channel open
         {tos,72}, % Deliver immediately
         {active,false}]). % Retrieve data from socket upon request
+
```
Comment (priestjim) [inline include/app.hrl]: Why? :-P

File: src/gen_rpc_acceptor.erl [hunk 2]
```diff
@@ -197,7 +197,10 @@ call_worker(Parent, WorkerPid, Ref, M, F, A) ->
     % and manifest as timeout. Wrap inside anonymous function with catch
     % will crash the worker quickly not manifest as a timeout.
     % See call_MFA_undef test.
-    % Fun = fun() -> catch erlang:apply(M, F, A) end,
-    PacketBin = erlang:term_to_binary({WorkerPid, Ref, catch erlang:apply(M, F, A)}),
+    Ret = case catch erlang:apply(M, F, A)of
```
Comment (priestjim) [inline src/gen_rpc_acceptor.erl]: Space at `of`

File: src/gen_rpc_acceptor.erl [hunk 3]
```diff
@@ -197,7 +197,10 @@ call_worker(Parent, WorkerPid, Ref, M, F, A) ->
...
+               {'EXIT', _} = V -> {badrpc, V};
```
Comment (priestjim) [inline src/gen_rpc_acceptor.erl]: This should return badrpc for errors too! Look:

    rpc:call(node(), erlang, error, [test]).
    {badrpc,{'EXIT',{test,[{rpc,local_call,3,
                            [{file,"rpc.erl"},{line,330}]},
                       {erl_eval,do_apply,6,[{file,"erl_eval.erl"},{line,673}]},
                       {shell,exprs,7,[{file,"shell.erl"},{line,686}]},
                       {shell,eval_exprs,7,[{file,"shell.erl"},{line,641}]},
                       {shell,eval_loop,3,[{file,"shell.erl"},{line,626}]}]}}}

File: .travis.yml [hunk 4]
```diff
@@ -16,6 +16,7 @@ script: "make test xref dialyzer"
 branches:
   only:
     - master
+    - presjimbench
```
Comment (priestjim) [inline .travis.yml]: That's not needed, I'll update the travis branches when I implement the perftest!

File: include/app.hrl (hunk 1 above)
Comment (linearregression) [inline include/app.hrl]: Something is weird is happening on my system. I need to clean my system up.

Comment (linearregression) [inline include/app.hrl]: Now I see the cfd282b was removeing that type socket_setopt(). But this is a long string of checking an this does not exist at the begining. Git "summarize" the changes and end up the difference is a new line.

File: src/gen_rpc_acceptor.erl (hunk 2 above)
Comment (linearregression) [inline src/gen_rpc_acceptor.erl]: This part is a direct rip off from the rpc src code :D.
They put {badrpc, {'EXIT, Stuff}} to match rpc(someothernode, F, A).

Comment (priestjim) [inline src/gen_rpc_acceptor.erl]: I think our way is cleaner though. Do you see anything wrong with what I suggested?

Comment (linearregression) [inline src/gen_rpc_acceptor.erl]: This PR is too messed up. After a good night sleep, I submit PR#18. It looks like rpc is hiding throw.

**Pull request closed without merging**